test/index.js

```diff
@@ -14,12 +14,24 @@ test('parse selector', (t) => {
   t.end()
 })
 
-test('parse selector class with attributes', (t) => {
+test('parse selector class with attributes class as string', (t) => {
   const obj = parse(['p.italic', { class: 'bold' }])
   t.equal(obj.attrs.class, 'italic bold')
   t.end()
 })
 
+test('parse selector class with attributes class as array', (t) => {
+  const obj = parse(['p.italic', { class: ['bold'] }])
+  t.deepEqual(obj.attrs.class, ['italic', 'bold'])
```

Comment:
I guess I see now when you originally said that this could be a lot of extra code. If you don't want this to be in `parse-hyperscript`, and rather this should only be done in `preact-hyperscript` if it's the one that supports `classNames` then for this very specific use case (classname as array) it might require doing the work of this PR in `preact-hyperscript` itself (and even then probably here as well). Ah well, let me know how you wanna proceed